ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -455,6 +481,9 @@ private void close(SelectionKey key) {
       activeConnections.set(this.keyMap.size());
       try {
         transmission.close();
+        if (pendingHandshakeTransmissions.get(transmission.getConnectionId()) != null) {
+          pendingHandshakeTransmissions.remove(transmission.getConnectionId());
+        }
       } catch (IOException e) {
```

Comment:
1. If a key can be closed in a different thread, would this part be thread safe?

2. Not related to this PR.
`Transmission.close()` will NOT throw an `IOException` if taking a look at its implementation in both `PlainTextTransmission` and `SSLTransmission`  though the abstract method defines so.